google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateQuery.java

```diff
@@ -224,7 +222,11 @@ RunAggregationQueryRequest toProto(@Nullable final ByteString transactionId) {
       } else {
         throw new RuntimeException("Unsupported aggregation");
       }
-      aggregation.setAlias(aggregateField.getAlias());
+      // Map all client-side aliases to a unique short-form alias.
+      // This avoids issues with client-side aliases that exceed the 1500-byte string size limit.
+      String serverAlias = "aggregate_" + aggregationNum++;
+      aliasMap.put(serverAlias, aggregateField.getAlias());
+      aggregation.setAlias(serverAlias);
       aggregations.add(aggregation.build());
```

Comment:
will the deduplication still work?

Reply:
My understanding is:
duplicate aliases *are not* allowed
duplicate aggregates *are* allowed

is that right @cherylEnkidu @MarkDuckworth ?

this code uses `aggregate_<counter>` as alias, so there should not be any duplicate aliases.

also, there's a test named `canGetDuplicateAggregations` below which currently passes against the emulator.

Reply:
ah... it's coming back to me now... I believe we decided to remove duplicate aggregates even though they are allowed to improve performance.

Reply:
Done.

Followed what was done in Android. PTAL.

Reply:
LGTM